src/rpc/util.h

```diff
@@ -399,6 +405,7 @@ class RPCHelpMan
     const std::vector<RPCArg> m_args;
     const RPCResults m_results;
     const RPCExamples m_examples;
+    mutable const JSONRPCRequest* m_req{nullptr}; // A pointer to the request for the duration of m_fun()
```

Comment:
In commit "rpc: Add Arg() default helper" (fa6b2da57ef7ff125a493c7835cb15935255ff8c)

Any reason to not just drop const from `HandleRequest` instead of making this variable mutable? Keeping the const there seems a little misleading

Reply:
> Also, the value will be nullptr before and after the call to HandleRequest

Does this also hold if `m_fun` throws an exception?

Reply:
@ajtowns I raised a similar concern, but it looks like this is safe: https://github.com/bitcoin/bitcoin/pull/28230#discussion_r1285851447